mappings/net/minecraft/network/encryption/NetworkEncryptionUtils.mapping

```diff
@@ -1,49 +1,229 @@
 	METHOD method_15234 decryptSecretKey (Ljava/security/PrivateKey;[B)Ljavax/crypto/SecretKey;
+		COMMENT Decrypts RSA-encrypted AES secret key.
+		COMMENT
+		COMMENT @throws NetworkEncryptionException when the secret key is malformed
 		ARG 0 privateKey
 		ARG 1 encryptedSecretKey
 	METHOD method_15235 cipherFromKey (ILjava/security/Key;)Ljavax/crypto/Cipher;
+		COMMENT Creates an AES cipher from the key.
+		COMMENT
+		COMMENT @throws NetworkEncryptionException when creation fails, e.g. due to invalid key
 		ARG 0 opMode
 		ARG 1 key
+			COMMENT the AES secret key
 	METHOD method_15236 hash ([[B)[B
+		COMMENT {@return the SHA-1 hash of {@code bytes}}
 		ARG 0 bytes
 	METHOD method_15237 generateServerKeyPair ()Ljava/security/KeyPair;
+		COMMENT {@return a new {@value RSA_KEY_LENGTH} bit RSA public/private key pair}
+		COMMENT
+		COMMENT @throws NetworkEncryptionException when generation fails
 	METHOD method_15238 encrypt (Ljava/security/Key;[B)[B
+		COMMENT Encrypts a data. The algorithm is determined from the key used.
+		COMMENT
+		COMMENT @throws NetworkEncryptionException when encryption fails, e.g. due to invalid key
+		COMMENT
+		COMMENT @see #decrypt(Key, byte[])
 		ARG 0 key
+			COMMENT encryption key (e.g. AES secret key or RSA public key)
 		ARG 1 data
-	METHOD method_15239 generateKey ()Ljavax/crypto/SecretKey;
-	METHOD method_15240 generateServerId (Ljava/lang/String;Ljava/security/PublicKey;Ljavax/crypto/SecretKey;)[B
+	METHOD method_15239 generateSecretKey ()Ljavax/crypto/SecretKey;
+		COMMENT {@return a new {@value AES_KEY_LENGTH} bit AES secret key}
+		COMMENT
+		COMMENT @throws NetworkEncryptionException when generation fails
+	METHOD method_15240 computeServerId (Ljava/lang/String;Ljava/security/PublicKey;Ljavax/crypto/SecretKey;)[B
+		COMMENT {@return the computed server ID}
+		COMMENT
+		COMMENT @implNote The server ID is a SHA-1 hash of ISO-8859-1 encoded {@code baseServerId},
+		COMMENT {@code publicKey}, and {@code secretKey}.
+		COMMENT
+		COMMENT @throws NetworkEncryptionException when computation fails
 		ARG 0 baseServerId
 		ARG 1 publicKey
 		ARG 2 secretKey
-	METHOD method_15241 crypt (ILjava/lang/String;Ljava/security/Key;)Ljavax/crypto/Cipher;
+	METHOD method_15241 createCipher (ILjava/lang/String;Ljava/security/Key;)Ljavax/crypto/Cipher;
+		COMMENT Creates a cipher to perform encryption or decryption operation.
 		ARG 0 opMode
 		ARG 1 algorithm
 		ARG 2 key
-	METHOD method_15242 readEncodedPublicKey ([B)Ljava/security/PublicKey;
-		ARG 0 bytes
+	METHOD method_15242 decodeEncodedRsaPublicKey ([B)Ljava/security/PublicKey;
+		COMMENT Decodes a X.509-encoded RSA public key.
+		COMMENT
+		COMMENT @throws NetworkEncryptionException when the key is malformed
+		COMMENT
+		COMMENT @see #decodeRsaPublicKeyPem(String)
+		ARG 0 key
 	METHOD method_15243 decrypt (Ljava/security/Key;[B)[B
+		COMMENT Decrypts an encrypted data. The algorithm is determined from the key used.
```

Comment:
What is 'an encrypted data' in this context?

Reply:
Any data (passed as byte[]).